### PR TITLE
Integrate geolocation/geocoding into EventCreator

### PR DESCRIPTION
**Pull request by Ef55**
This also contains PR #70:
This one improve the UI built in it, and will be merged at the same time to reduce testing effort.

### CODE REVIEW

**Comment by Ef55** [comment]
Current look:

<details> 
<summary>Main UI: </summary>

</details>

<details> 
<summary>Main UI (with geolocation enabled): </summary>

</details>

<details> 
<summary>Geolocation tab: </summary>

</details>

<details> 
<summary>Geolocation tab (with completed query): </summary>

</details>

File: app/src/androidTest/java/com/github/onedirection/EventCreatorTest.java [hunk 1]
```diff
@@ -91,72 +142,95 @@ public void verifyEventViewIsCorrectlyCalled() {
...
     }
+
+
+    @Test
+    public void phoneLocationCanBeUsedWithoutUI() throws ExecutionException, InterruptedException {
+        final DeviceLocationProvider[] testClass = new DeviceLocationProvider/*The array is so that it works*/[1];
+        ActivityScenario.launch(EventCreator.class).onActivity(activity -> testClass[0] = (DeviceLocationProvider)activity);
+        testClass[0].startLocationTracking().get();
+        assertThat(testClass[0].getNextLocation().get(), not(nullValue()));
+    }
```
Comment (Ef55) [inline app/src/androidTest/java/com/github/onedirection/EventCreatorTest.java:234]: Ok, I won't even comment on the array usage, but I cannot even begin to fathom why you would put that comment there.

**Comment by Ef55** [comment]
<details> 
<summary>For those wondering "how did you disable some tests while still preserving decent coverage ?": </summary>

</details>

File: app/src/main/java/com/github/onedirection/events/EventCreatorGeolocationFragment.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+            validate.setEnabled(valid);
+        });
+
+        // Click listeners
+        getView().findViewById(R.id.buttonSearchLocation).setOnClickListener(v ->
+            setCurrentRequest(
+                    generateGeocodingRequest(locationQuery.getText().toString())
+            )
+        );;
```
Comment (Sosol02) [inline app/src/main/java/com/github/onedirection/events/EventCreatorGeolocationFragment.java:97]: Nitpick: extra ';'

Comment (Sosol02) [inline app/src/main/java/com/github/onedirection/events/EventCreatorGeolocationFragment.java:97]: Nitpick: extra ';'

File: app/src/main/java/com/github/onedirection/events/EventCreatorViewModel.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        this.endTime = new MutableLiveData<>(event.getEndTime());
+        this.useGeolocation = new MutableLiveData<>(event.getCoordinates().isPresent());
+
+        this.eventId = event.getId();
+        this.isEditing = true;
+        this.idling = new CountingIdlingResource("Event creator is loading.");
+    }
+
+    public Event generateEvent(){
+        // TODO: use coordinates
```
Comment (justagoodguy-omega) [inline app/src/main/java/com/github/onedirection/events/EventCreatorViewModel.java:68]: You can delete this I believe

**Review by Sosol02 — APPROVED** [review]
lgtm

File: app/src/main/java/com/github/onedirection/geolocation/DeviceLocationProvider.java [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+        }
+        else{
+            return CompletableFuture.completedFuture(getLastLocation());
+        }
+    }
+
+    private final static LocationRequest LOCATION_REQUEST = LocationRequest.create();
+    static {
+        LOCATION_REQUEST.setInterval(10000);
+        LOCATION_REQUEST.setFastestInterval(5000);
```
Comment (Flechman) [inline app/src/main/java/com/github/onedirection/geolocation/DeviceLocationProvider.java:72]: Magic number, could be good to create a private static variable

**Review by Flechman — APPROVED** [review]
The PR could've been splitted in two smaller PRs, but overall provides great functionality to the app

**Pull request merged**